src/sonic-config-engine/portconfig.py

```diff
@@ -157,7 +157,11 @@ def gen_port_config(ports, parent_intf_id, index, alias_at_lanes, lanes, k,  off
             ports[intf_name]['alias'] = alias_at_lanes.split(",")[alias_start]
             ports[intf_name]['lanes'] = ','.join(lanes.split(",")[alias_start:alias_start+step])
             if speed:
-                ports[intf_name]['speed'] = speed
+		if 'G'  not in speed.upper():
```

Comment:
I think the logic should be:
1,  All numbers should be fine.
2, If it has G at the end,  replace it with 000
3, Any character other than "G" should fail here.

Reply:
addressed